### DISCUSSION TITLE
Window manager poll

### DISCUSSION THREAD

**Discussion by stefonarch** [lxqt/lxqt#2166] [Show and tell]
![schermata-12-20-11-01](https://user-images.githubusercontent.com/10681413/146749658-62c8e569-f204-4dec-a8cf-5fd0587bf44f.png)

https://twitter.com/lxqt_project/status/1472307610426789891

**Comment by dglent** [comment]
I use Openbox (i don't have twitter account!)

**Reply by tsujan** [reply]
I use KWin and, sometimes — just for the fun of it — compiz-reloaded. (I have no twitter account either.)

**Reply by stefonarch** [reply]
And me xfwm4, but voted already ;)

**Reply by ssrublev** [reply]
Voted =))

**Reply by elviosak** [reply]
Xfwm4, also no twitter
Tried kwin  and it feels a bit slow, also compositing leaves artifacts(?) (a lot of colored squares of different colors) after suspend/wake... Tried a bunch of other wms and some almost worked for me, but had issues probably because of weird monitor setup..

**Reply by ssrublev** [reply]
I use MATE+Compiz for many years since Gnome2 times, before I was at Gnome2 in Ubuntu. No Unity, KDE, Gnome3, XFCE and Cinnamon made me switch. My main system is Fedora-MATE-Compiz for decade. 

I recently installed Lubuntu in Ubuntu from command line. I saw LXDE desktops previously on flash-live systems. I wasn't impressed very much, but I was very impressed by LXQt in 2021. That's why I now have it on my main Fedora system (command line install also).

In Lubuntu and Fedora the Openbox installed itself as default with LXQt. I didn't try Compiz with it. Maybe I should try. But I like the Picom. In Ubuntu, it calls itself Compton.

**Reply by tsujan** [reply]
> I didn't try Compiz with it.

If by compiz you mean compiz-reloaded (0.8.X), it works fine, provided that you find the correct graphic settings. For example, it was choppy here with modesetting+DRI3 but very smooth with Intel driver + DRI2. I don't think there's any difference between MATE and LXQt in this regard.

**Reply by ssrublev** [reply]
BTW I also found that Candy icons https://store.kde.org/p/1305251 look very good with LXQt Kvantum theme. In Fedora, they are packaged as "candy-icon-theme".

**Reply by Faehnchen** [reply]
I spend so much time in the openbox config shipped with lubuntu 20.04 getting my DE looking and feeling really nice and dapper, I now like it too much to switch to another one immediatly. So I can't really compare to the other WMs. The comments inside the config file make it very easy to understand the functions as a newbee and implement one's own settings. 
However I also aggree with @ssrublev over the good looks of Candy-Icons especially with dark theming like Arc-dark. It's just a minor annoyance that some Qt-applications aren’t yet present in the icon sets of the big distros and stick a bit out with their default icons (hope that changes sometime). So my solution was to edit the desktop config files in /usr/share/applications to match the icon-instructions roughly with their GTK-counterparts so Qlipper > Glipper; Featherpad > Leafpad and so on. Please excuse this heresy! :p

**Reply by ssrublev** [reply]
> If by compiz you mean compiz-reloaded (0.8.X), it works fine, provided that you find the correct graphic settings. For example, it was choppy here with modesetting+DRI3 but very smooth with Intel driver + DRI2. I don't think there's any difference between MATE and LXQt in this regard.

Thank you very much for suggesting Compiz. I am used to it and it's good that it works with LXQt, but sometimes buggy. Compiz is buggy, I have 0.8.18 with LXQt 0.17.0. Is there a way to get LXQt 1.0 in Fedora or Ubuntu? I think if I get Compiz tuned propely (there are so many settings), I will not use Openbox much.

**Reply by stefonarch** [reply]
> Is there a way to get LXQt 1.0 in Fedora or Ubuntu?

Uninstalling everything and compiling from source, as downside this bypasses your package manager. Tested in Ubuntu 21.10 in VM, works fine.

https://github.com/lxqt/lxqt/wiki/Building-from-source

**Reply by tsujan** [reply]
> Compiz is buggy, I have 0.8.18 with LXQt 0.17.0

Using Compiz isn't easy, not because it's buggy — of course, it has bugs but not more than other programs of its kind — but for two reasons: (1) It needs recompilation once in a while, when the system is upgraded (which happens often in Arch/Manjaro); and (2) the user needs to find the best graphical settings with it (for me, Intel driver + DRI2).

As far as I've seen, SDDM doesn't like the graphical settings that I use with Compiz (strange flashes in the login screen) but its problem is tolerable and, after all, the problem belongs to SDDM. KDE has worse problems with those settings.

I can't recommend Compiz, for the reasons mentioned above. It's just fun for those who can handle its hassles. It feels smoother and livelier than KWin when it works fine.

**Reply by ssrublev** [reply]
> Is there a way to get LXQt 1.0 in Fedora or Ubuntu? 

I have found a way to get LXQt 1.0 in Fedora by installing Fedora Rawhide. As of 12 February 2022, LXQt 1.0 is supplied via Fedora Rawhide repository and I am now able to test it (Rawhide is a test repository subject to bugs).

**Reply by hngts** [reply]
Openbox is pure gold. No match for that one. In love with openbox since Crunchbang linux and have no plans changing anything. Kill openbox and You killed me.

**Reply by wishborn79** [reply]
openbox ftw

**Comment by stefonarch** [comment]
2 days maybe was too short, anyway, some idea we can get:

![schermata-12-21-22-59](https://user-images.githubusercontent.com/10681413/147003126-135ae357-6437-4de8-aaeb-c09c35de515c.png)

**Reply by tsujan** [reply]
The reason may be twofold: (1) Openbox still works and is easy to use; and (2) Most distros ship LXQt with Openbox. But its bugs (which do exist) won't be fixed because its development stopped years ago.

Xfwm4 is light, configurable and easy to use. Moreover, it comes with its own compositor.

Sadly, KWin has too many (artificial) KDE dependencies; otherwise, it would be ideal.

**Reply by Conan-Kudo** [reply]
There's also KWinFT, I suppose, which I think @subdiff develops. That one aims to reduce the KDE/KF5 dependencies, I think?

**Reply by tsujan** [reply]
> There's also KWinFT

→ https://github.com/lxqt/lxqt/discussions/2144#discussioncomment-1768272

**Reply by subdiff** [reply]
@Conan-Kudo Thanks for the shout out. My current plan is to make KDE/KF5 dependencies optional in KWinFT or rather in the libraries that I'll split out from it in the next months.

Depending on the level of functionality users expect from the windowing manager the system designer should then to be able to compile only the functionality (and its dependencies) into the window manager that are required. So for example if your users won't miss windowing rules in their WM, then you can exclude this subsystem from compilation.

And once this kind of system has been fleshed out some more it should be possible to comfortably create a WM similar in footprint and functionality to Openbox with the KWinFT libraries too.

**Reply by redtide** [reply]
I also really like Openbox, it would be great if a Wayland "version" like labwc could be used.

**Comment by ringo32** [comment]
normally i used xfwm4 but nowjust fulltime openbox , hadsome issues with meta+1 dont know why have to see if its on my side .. probably....kwin does not like azerty thats a pitty instead é it makes more like a E with a ' above pitty :)

**Comment by Po150N** [comment]
can you create a specific window manager for lxqt?

**Comment by tsujan** [comment]
Wow! While KWin has got issues like [this](https://github.com/lxqt/pcmanfm-qt/issues/1580#issuecomment-1121451363) and [this](https://github.com/lxqt/pcmanfm-qt/issues/1580#issuecomment-1121928116), KWinFT doesn't have those regressions. Moreover, desktop cube is available only with KWinFT.

**Reply by Chiitoo** [reply]
Not entirely sure if it's related to 'desktop cube', but there's some talk (a little dated by now) about why the cube family of effects was dropped here:

https://blog.martin-graesslin.com/blog/2021/11/evolving-3d-desktop-effects-in-plasma/

Perhaps they'll make a return in a Qt Quick form at some point... which is what they also suggest for re-implementing the desktop geometry effect with too, which I used a lot, and which was also removed not too long ago...

**Reply by tsujan** [reply]
Yes, I'd read it. *This* is a good place to use Qt Quick, not in GUI apps, where it breaks platform integration.

**Reply by stefonarch** [reply]
Another article: https://blog.vladzahorodnii.com/2022/05/09/whats-cooking-in-kwin-plasma-5-25/

**Comment by edwardp3** [comment]
I just switched from openbox to xfwm4 for the translucency. Tried (and uninstalled) picom, it ran too slow on this hardware.
